snuba/subscriptions/executor.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import annotations
+
+from concurrent.futures import Future, ThreadPoolExecutor
+
+from snuba import settings
+from snuba.api.query import ClickhouseQueryResult, parse_and_run_query
+from snuba.datasets.dataset import Dataset
+from snuba.request import Request
+from snuba.subscriptions.consumer import Tick
+from snuba.subscriptions.data import Subscription
+from snuba.subscriptions.scheduler import ScheduledTask
+from snuba.utils.metrics.timer import Timer
+
+
+executor_pool = ThreadPoolExecutor(max_workers=settings.SUBSCRIPTIONS_MAX_CONCURRENT_QUERIES)
+
+
+class Executor:
+    """
+    Responsible for running subscription queries for a given `Dataset`. Each call to
+    `execute` should be passed the `ScheduledTask[Subscription]` to run and the `Tick`
+    we're running the query for. The `Subscription` generates the `Request` object used
+    to make the query, and the result is then returned as a `Future`.
+    """
+
+    def __init__(self, dataset: Dataset):
```

Comment:
```suggestion
    def __init__(self, dataset: Dataset) -> None:
```

Reply:
Do we really need to specify the return value of constructors?

Reply:
~Sometimes mypy complains if they are not defined, I haven't tracked down when/why (not sure if that was a prior version thing, or if it only needs to be done in certain circumstances) — it seems easier just to specify it and avoid any questions.~

Ok, you can disregard this, unless you'd like to add it for consistency; https://mypy.readthedocs.io/en/latest/class_basics.html#annotating-init-methods